internal/pkg/cli/deploy/workload_test.go

```diff
@@ -204,6 +205,21 @@ func TestWorkloadDeployer_UploadArtifacts(t *testing.T) {
 		wantBuildRequired bool
 		wantErr           error
 	}{
+		"error if docker engine is not running": {
+			inMockUserTag: "v1.0",
+			inDockerBuildArgs: map[string]*manifest.DockerBuildArgs{
+				"mockWkld": {
+					Dockerfile: aws.String("mockDockerfile"),
+					Context:    aws.String("mockContext"),
+				},
+			},
+			mock: func(t *testing.T, m *deployMocks) {
+				m.mockdockerEngineRunChecker.EXPECT().CheckDockerEngineRunning().Return(errors.New(`docker daemon is not responsive:
```

Comment:
nit: for the test case, we're just making sure that calling `UploadArtifacts()` returns an error when `CheckDockerEngineRunning` returns an error. it doesn't have to be a "real" error 😊 
```suggestion
				m.mockdockerEngineRunChecker.EXPECT().CheckDockerEngineRunning().Return(errors.New("an error")
```